test/src/core/testCreateSession.js

```diff
@@ -432,4 +432,255 @@ describe('test Request With Credentials', function () {
+          'accessKey': 'test_key',
+          'os': 'OS X',
+          'osVersion': 'Monterey',
+          'buildName': 'nightwatch-test-build'
+        },
+        'build': 'Nightwatch Programmatic Api Demo',
+        'chromeOptions': {
+          'w3c': false
+        },
+        'goog:chromeOptions': {
+          'w3c': false
+        },
+        'name': 'Try 1'
+      }
+    });
+  });
+
+  it('Test create session with browserstack and browserName set to null', async function() {
+    nock('https://hub.browserstack.com')
+      .post('/wd/hub/session')
+      .reply(201, function (uri, requestBody) {
+        const reqObj = JSON.parse(requestBody);
+
+        assert.deepEqual(reqObj, {
+          'desiredCapabilities': {
+            'browserName': null,
+            'bstack:options': {
+              'local': 'false',
+              'userName': 'test_user',
+              'accessKey': 'test_key',
+              'os': 'OS X',
+              'osVersion': 'Monterey',
+              'buildName': 'nightwatch-test-build'
+            },
+            'chromeOptions': {
+              'w3c': false
+            },
+            'name': 'Try 1',
+            'build': 'Nightwatch Programmatic Api Demo'
+          },
+          'capabilities': {
+            'alwaysMatch': {
+              'bstack:options': {
+                'local': 'false',
+                'userName': 'test_user',
+                'accessKey': 'test_key',
+                'os': 'OS X',
+                'osVersion': 'Monterey',
+                'buildName': 'nightwatch-test-build'
+              }
+            }
+          }
+        });
+
+        return {
+          value: {
+            sessionId: '1352110219202',
+            capabilities: reqObj.desiredCapabilities
+          }
+        };
+      });
+
+
+    nock('https://api.browserstack.com')
+      .get('/automate/builds.json')
+      .reply(200,  [
+        {
+          automation_build: {
+            name: 'WIN_CHROME_PROD_SANITY_LIVE_1831',
+            duration: 47,
+            status: 'running',
+            hashed_id: '8dd73aad3365429dec0ec12cf64c0c475a22dasds',
+            build_tag: null
+          }
+        },
+        {
+          automation_build: {
+            name: 'External monitoring - aps - 2022-08-30',
+            duration: 44,
+            status: 'done',
+            hashed_id: '8dd73aad3365429dec0ec12cf64c0c475a22dasdk',
+            build_tag: null
+          }
+        } 
+      ]);
+    
+    const client = Nightwatch.createClient({
+      webdriver: {
+        start_process: false
+      },
+      selenium: {
+        host: 'hub.browserstack.com',
+        port: 443
+      },
+      desiredCapabilities: {
+        'bstack:options': {
+          local: 'false',
+          userName: 'test_user',
+          accessKey: 'test_key',
+          os: 'OS X',
+          osVersion: 'Monterey'
+        },
+        browserName: null,
+        chromeOptions: {
+          w3c: false
+        }
+      },
```

Comment:
I think we should use mobile capabilties in this test as `browsername: null` will be used for mobile capabilites.

Reply:
updated